dist/pom.xml

```diff
@@ -409,6 +409,42 @@
                     </excludes>
                 </configuration>
             </plugin>
+            <plugin>
+                <!-- The dependency scopes of spark-rapids-jni and jucx are `provided` in aggregator submodule,
+                 this will skip the dedupe processing in `create-parallel-world`.
+                 Unpack spark-rapids-jni and jucx jars to `parallel-world` directory after `create-parallel-world`,
+                 then `parallel-world` will be packaged.
+                 Note all Shims use the same version of spark-rapids-jni and jucx -->
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>unpack-spark-rapids-jni-and-ucx</id>
+                        <!-- run after `create-parallel-world`, the phase equals to the phase of the `create-parallel-world` -->
+                        <phase>generate-resources</phase>
+                        <goals>
+                            <goal>unpack</goal>
```

Comment:
Rather than manually maintaining a dependency list here that needs to be sync'd with the same list in the aggregator pom, would it be possible to automatically pull in the immediate provided dependences using the `unpack-dependencies` goal with configs like the following?
```
  <excludeTransitive>true</excludeTransitive>
  <includeScope>provided</includeScope>
```

Reply:
`excludeTransitive` should be false because we want the transitive provided jars for the `dist` submodule.

Tested
```
        <goals>
            <goal>unpack-dependencies</goal>
        </goals>
        <configuration>
            <excludeTransitive>false</excludeTransitive>
            <includeScope>provided</includeScope>
        </configuration>
```

It did not work.

```
cd dist
mvn dependency:tree

[INFO] com.nvidia:rapids-4-spark_2.12:jar:22.06.0-SNAPSHOT
[INFO] \- com.nvidia:rapids-4-spark-aggregator_2.12:jar:spark311:22.06.0-SNAPSHOT:compile
[INFO]    +- com.nvidia:rapids-4-spark-sql_2.12:jar:spark311:22.06.0-SNAPSHOT:compile
[INFO]    |  +- com.nvidia:spark-rapids-jni:jar:cuda11:22.06.0-SNAPSHOT:compile
[INFO]    |  |  \- org.slf4j:slf4j-api:jar:1.7.30:compile
[INFO]    |  +- com.nvidia:rapids-4-spark-common_2.12:jar:22.06.0-SNAPSHOT:compile
[INFO]    |  \- com.google.flatbuffers:flatbuffers-java:jar:1.11.0:compile
[INFO]    +- com.nvidia:rapids-4-spark-shuffle_2.12:jar:spark311:22.06.0-SNAPSHOT:compile
[INFO]    |  \- org.openucx:jucx:jar:1.12.1:compile
[INFO]    \- com.nvidia:rapids-4-spark-udf_2.12:jar:spark311:22.06.0-SNAPSHOT:compile
[INFO] ------------------------------------------------------------------------

cd aggregator
mvn dependency:tree

[INFO] com.nvidia:rapids-4-spark-aggregator_2.12:jar:22.06.0-SNAPSHOT
[INFO] +- com.nvidia:rapids-4-spark-sql_2.12:jar:spark311:22.06.0-SNAPSHOT:compile
[INFO] |  +- com.nvidia:rapids-4-spark-common_2.12:jar:22.06.0-SNAPSHOT:compile
[INFO] |  \- com.google.flatbuffers:flatbuffers-java:jar:1.11.0:compile
[INFO] +- com.nvidia:rapids-4-spark-shuffle_2.12:jar:spark311:22.06.0-SNAPSHOT:compile
[INFO] +- com.nvidia:rapids-4-spark-udf_2.12:jar:spark311:22.06.0-SNAPSHOT:compile
[INFO] +- org.openucx:jucx:jar:1.12.1:provided
[INFO] \- com.nvidia:spark-rapids-jni:jar:cuda11:22.06.0-SNAPSHOT:provided
[INFO]    \- org.slf4j:slf4j-api:jar:1.7.30:provided
[INFO] ------------------------------------------------------------------------


```

The output of `dependency:tree`  in `dist` is:
[INFO]    |  \- org.openucx:jucx:jar:1.12.1:compile

While the output of ``  in `aggregator` is:
[INFO] +- org.openucx:jucx:jar:1.12.1:provided

The `jucx` is provided in aggregator.xml.
Seems weird.

Currently, we only have 2 external jars to handle.
IMO we now can explicitly specify the external jars, although there are 2 places to maintain.